redbpf/src/lib.rs

```diff
@@ -820,6 +832,43 @@ impl<K: Clone, V: Clone> Iterator for MapIter<'_, '_, K, V> {
     }
 }
 
+impl StackTrace<'_> {
+    pub fn new(map: &Map) -> StackTrace<'_> {
+        StackTrace {
+            base: map
+        }
+    }
+
+    pub fn get(&mut self, id: libc::c_int) -> Option<BpfStackFrames> {
+        unsafe {
+            let mut value = MaybeUninit::uninit();
+
+            let ret = bpf_sys::bpf_lookup_elem(
+                self.base.fd,
+                &id as *const libc::c_int as *mut libc::c_int as _,
```

Comment:
An easier way to do this might be at https://github.com/redsift/redbpf/blob/78f6ac9113d9cfad44e4b4dccc06e1fba763332e/redbpf/src/lib.rs#L814 .